scene/gui/scroll_container.cpp

```diff
@@ -102,37 +102,42 @@ void ScrollContainer::gui_input(const Ref<InputEvent> &p_gui_input) {
 
 	if (mb.is_valid()) {
 		if (mb->get_button_index() == MouseButton::WHEEL_UP && mb->is_pressed()) {
-			// only horizontal is enabled, scroll horizontally
-			if (h_scroll->is_visible() && (!v_scroll->is_visible() || mb->is_shift_pressed())) {
+			// By default, the vertical orientation takes precedence. This is an exception.
+			if ((horizontal_scroll_mode != SCROLL_MODE_DISABLED && mb->is_shift_pressed()) || (!v_scroll->is_visible() && vertical_scroll_mode != SCROLL_MODE_SHOW_NEVER)) {
```

Comment:
`!v_scroll->is_visible() && vertical_scroll_mode != SCROLL_MODE_SHOW_NEVER`
This part seems to repeat a few times. It should be stored in a bool (unless there is a reason not to?).

Reply:
Because there are many branches, these statements may not be executed, but may be executed if saved ahead of time.

Reply:
I think it doesn't really matter if it's evaluated unnecessarily. It's just a getter call and a comparison. Unless it would affect the logic somehow?

Reply:
Currently unaffected by `scroll_value`.